docs/contributing-to-airbyte/README.md

```diff
@@ -72,6 +73,35 @@ The CDK currently does not support creating destinations, but it will very soon.
+|-----------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------|----------------|
+| Adding a required parameter to a connector's `spec`                                           | Users will have to add the new parameter to their `config`                                                       | Major          |
+| Changing a format of a parameter in a connector's `spec` from a single parameter to a `oneOf` | Users will have to edit their `config` to define their old parameter value in the `oneOf` format                 | Major          |
+| Removing a stream from a connector's `catalog`                                                | Data that was being synced will no longer be synced                                                              | Major          |
+| Renaming a stream in a connector's `catalog`                                                  | Users will have to update the name of the stream in their `catalog`                                              | Major          |
+| Removing a column from a stream in a connector's `catalog`                                    | Users will have to remove that column from their `catalog`, data that was being synced will no longer be synced  | Major          |
+| Renaming a column from a stream in a connector's `catalog`                                    | Users will have to update the name of the column in their `catalog`                                              | Major          |
+| Changing the datatype for a column of a stream in a connector's `catalog`                     | Users will have to update that data type in their `catalog`, data that was being synced will have changed format | Major          |
+| Adding a non-required parameter to a connector's `spec`                                       | Users will have the option to use the required parameter in the future                                           | Minor          |
+| Adding a stream in a connector's `catalog`                                                    | Additional data will be synced                                                                                   | Minor          |
+| Adding a column to a stream's schema in a connector's `catalog`                               | Additional data will be synced                                                                                   | Minor          |
+| Updating the format of the connector's `STATE`                                                | Incremental streams will automatically run a full refresh only for the next sync                                 | Patch          |
```

Comment:
I think this is a tricky one. Full refreshes can be quite painful. I would love to see connectors migrating the state if at all possible.

Reply:
That being said, I don't have a strong opinion it fits into the semantic versioning. So carry on.

Reply:
I would love to see that too! I think that's a good input into our conversation about improving the way we automate things, but at the moment, this is unfortunately the current outcome of making the change. 

Depending on how painful we think the experience of running a full refresh is, we could update our notion of majors:
> - Major: a version in which a change is made which requires manual intervention (update to config or configured catalog) for an existing connection to continue to succeed, or one in which data that was previously being synced will no longer be synced

to include things that force full refreshes. I kept our reasoning behind the categorization (`we have our own take on semantic versioning, focusing on maintaining the best user experience of using a connector`) pretty vague so that we can play around with it.